### PR TITLE
Route builder improvements

### PR DESCRIPTION
**Pull request by JamesNK**
- Return endpoint convention builder from route builder
- Register marker service and test in route builder
- Get unit tests running

### CODE REVIEW

**Comment by JamesNK** [comment]
// @jtattermusch

File: sample/Server/Greeter.cs [hunk 1]
```diff
@@ -3,6 +3,7 @@
 using Grpc.Core;
 using Greet;
 using Microsoft.Extensions.Logging;
+using Microsoft.AspNetCore.Http;
```
Comment (JunTaoLuo) [inline sample/Server/Greeter.cs]: What's this for?

File: src/Grpc.AspNetCore.Server/GrpcEndpointRouteBuilderExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+// limitations under the License.
+
+#endregion
+
+using System;
+using System.Collections.Generic;
+using Grpc.AspNetCore.Server.Internal;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Routing;
+using ProtobufServiceDescriptor = Google.Protobuf.Reflection.ServiceDescriptor;
```
Comment (JunTaoLuo) [inline src/Grpc.AspNetCore.Server/GrpcEndpointRouteBuilderExtensions.cs]: We are trying to remove the dependency on Google.Protobuf.

Comment (JamesNK) [reply src/Grpc.AspNetCore.Server/GrpcEndpointRouteBuilderExtensions.cs]: Brah, I know. Will fix it after https://github.com/grpc/grpc-dotnet/issues/21

**Pull request merged**